frontend/frontend.go

```diff
@@ -20,7 +19,13 @@ type Frontend interface {
 type FrontendLLBBridge interface {
 	Solve(ctx context.Context, req SolveRequest, sid string) (*Result, error)
 	ResolveImageConfig(ctx context.Context, ref string, opt llb.ResolveImageConfigOpt) (digest.Digest, []byte, error)
-	Exec(ctx context.Context, meta executor.Meta, rootfs cache.ImmutableRef, stdin io.ReadCloser, stdout, stderr io.WriteCloser) error
+	// Run will start a container for the given process with rootfs, mounts.
+	// `id` is an optional name for the container so it can be referenced later via Exec.
+	// `started` is an optional channel that will be closed when the container setup completes and has started running.
+	Run(ctx context.Context, id string, rootfs cache.Mountable, mounts []executor.Mount, process executor.ProcessInfo, started chan<- struct{}) error
+	// Exec will start a process in container matching `id`. An error will be returned
+	// if the container failed to start (via Run) or has exited before Exec is called.
+	Exec(ctx context.Context, id string, process executor.ProcessInfo) error
```

Comment:
@hinshun and I were chatting, we could remove the `Run/Exec` api and instead expose the `Executor/CacheManager` apis here with something like this:
```
func (s *llbBridge) Executor() (executor.Executor, err error) {
	w, err := s.resolveWorker()
	if err != nil {
		return nil, err
	}
	return w.Executor(), nil
}
```

@tonistiigi not sure if you had a preference to this approach.

Reply:
I guess I would prefer just
```
type FrontendLLBBridge interface {
   executor.Executor
}
```